### PR TITLE
Fix for Issue 5: MemberAPI should export default new MemberAPI

### PR DESCRIPTION
**Pull request by josemigallas**
The best solution I've found is to refactor MemberApi into a static class, since there is no need to instance it and it must keep track of the current id regardless of where it is used.

### CODE REVIEW

**Comment by brauliodiez** [comment]
We only need to make:

export default new MemberAPI();

on the memberAPI module, that's an easy way to get a single instance and
the app working fine (no need to pollute with static every member)

**Comment by brauliodiez** [comment]
Take a look to sample 09

import MemberEntity from './memberEntity'
import MembersMockData from './memberMockData'
import * as _ from 'lodash'
// Sync mock data API, inspired from:
// https://gist.github.com/coryhouse/fd6232f95f9d601158e4
class MemberAPI {
  private _idSeed : number;

  public constructor() {
    this._idSeed = 20;
  }

  //This would be performed on the server in a real app. Just stubbing in.
  private _clone (item) {
  return JSON.parse(JSON.stringify(item)); //return cloned copy so that the
item is passed by value instead of by reference
  };

  //This would be performed on the server in a real app. Just stubbing in.
  _generateId() : number {
  return this._idSeed++;
  };


  // Just return a copy of the mock data
  getAllMembers() : Array<MemberEntity> {
return this._clone(MembersMockData);
}

  getMemberById(id : number) : MemberEntity {
var member = _.find(MembersMockData, {id: id});
return this._clone(member);
}

  saveAuthor(member: MemberEntity) {
//pretend an ajax call to web api is made here
console.log('Pretend this just saved the author to the DB via AJAX
call...');

if (member.id != -1) {
var existingAuthorIndex = _.indexOf(MembersMockData,
_.find(MembersMockData, {id: member.id}));
MembersMockData.splice(existingAuthorIndex, 1, member);
} else {
//Just simulating creation here.
//The server would generate ids for new authors in a real app.
//Cloning so copy returned is passed by value rather than by reference.
member.id = this._generateId();
MembersMockData.push(this._clone(member));
}

return member;
}

}

export default new MemberAPI();

**Comment by josemigallas** [comment]
Updated. Nevertheless, I must humbly say that I don't see it as a real TS approach, looks to me more like JS.

File: 06 AJAX Call/src/components/members/membersPage.tsx [hunk 1]
```diff
@@ -24,9 +24,8 @@ export default class MembersPage extends React.Component<Props, State> {
 
    // Changing to componentDidMount to handle initial ajax request response
    public componentDidMount() {
-     var memberAPI : MemberAPI = new MemberAPI();
      //this.state.members = memberAPI.getAllMembers();
```
Comment (Nasdan) [inline 06 AJAX Call/src/components/members/membersPage.tsx]: Remove comment

**Comment by Nasdan** [comment]
Merged

**Pull request merged**